### PR TITLE
Add support for custom trait bounds

### PR DESCRIPTION
**Pull request by blefevre**
Custom trait bounds can be added with the `#[tuple_types_custom_trait_bound(NewBound)` attribute.
This will use `NewBound` as the type for each Tuple component instead of the Trait being implemented.

### CODE REVIEW

File: src/semi_automatic.rs [hunk 1]
```diff
@@ -505,16 +494,45 @@ impl<'a> ToTupleImplementation<'a> {
...
+        let trait_ = if let Some(pos) = res
+            .attrs
+            .iter()
+            .position(|a| a.path.is_ident(TUPLE_TYPES_CUSTOM_TRAIT_BOUND))
+        {
+            let attr = &res.attrs[pos];
+            let trait_name = match attr.parse_meta().map_err(|e| {
+                Error::new(
+                    e.span(),
+                    "Invalid trait bound, advanced trait bounds are unsupported",
```
Comment (bkchr) [inline src/semi_automatic.rs:514]: Can we add support for this?

**Review by bkchr — COMMENTED** [review]
Hey, thank you! That is actually a nice idea.

I currently worked around this by using `tuple_types_no_default_trait_bound` attribute combined with writing the bounds on my own. 

If you add support for "advanced bounds", I would be really happy and merge it :)

**Comment by blefevre** [comment]
Supporting advanced trait bounds was simpler than expected so that's been added.

**Review by bkchr — APPROVED** [review]
Looks good ty!

**Pull request merged**